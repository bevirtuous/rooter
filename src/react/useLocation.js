import useStore from './store';

function useLocation() {
  const { current, routes } = useStore();

  return routes[current];
}

export default useLocation;
