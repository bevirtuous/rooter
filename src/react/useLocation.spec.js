import { renderHook } from '@testing-library/react-hooks';
import useLocation from './useLocation';

describe('React - useLocation', () => {
  it('should return current location', () => {
    const { result } = renderHook(() => useLocation());

    expect(result.current).toMatchSnapshot();
  });
});
