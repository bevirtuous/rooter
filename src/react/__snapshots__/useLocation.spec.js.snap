// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React - useLocation should return current location 1`] = `
Object {
  "fragment": "",
  "id": "default",
  "pathname": "/myroute/123",
  "search": "",
  "state": Object {},
}
`;
